### ISSUE TITLE
A11Y: Enable screenreader to read payment option before submitting the form in checkout when using keyboard navigation

### ISSUE THREAD

**Issue by MHB-dotSource** [shopware/platform#9346]
### Shopware Version

6.6.8

### Affected area / extension

Platform(Default)

### Actual behaviour

When navigating the payment options in the checkout using the arrow keys, the payment method is selected immediately and the page is reloaded. If a screen reader is in use, the screen reader is not able to read the payment method text before the page is reloaded.

### Expected behaviour

- the screenreader is able to read the payment option text
- the user can select the option wich triggers a page reload or go to the next option using the arrow keys

### How to reproduce

- have at least 2 payment methods available for the storefront
- go to checkout payment step
- activate a screenreader
- navigate to the payment options using the keyboard
- go to the next payment option using the arrow keys

**Comment by mstegmeyer** [comment]
@MHB-dotSource We have significantly reworked the payment selection behind the ACCESSIBILITY_TWEAKS feature flag on Shopware 6.6.10. Mainly, all payment methods are now shown instead of just the first 5 and the re-ordering upon selection does not happen again. Does this meet your expectations then or do you think this is not sufficient?

**Comment by MHB-dotSource** [comment]
@mstegmeyer That is good first step. I think it is not yet sufficient for a screen reader user that also uses the keyboard navigation. If I understand correctly, the page reload still happens as soon as a new payment method is selected. So the user does not have the chance to read and understand the option they selected, before the page reload triggers.

**Comment by sguder** [comment]
Tested with NVDA screen reader. (Expected a german website obviously)

https://github.com/user-attachments/assets/1b7c9343-7516-431c-82f3-3cd22f3925c7

**Comment by mstegmeyer** [comment]
@ravenokko any clue, if we should do more? We're not sure how to do this with that auto-submit form.

**Comment by mstegmeyer** [comment]
The "bad" thing that we do is reloading the page as soon as another radio is selected. That should ideally not happen. We try to make it as tolerable as possible by restoring the focus again so it more or less feels as if you would cycle through the options.

For now, we will not take further action here.
